splinepy/multipatch.py

```diff
@@ -0,0 +1,358 @@
+
+        Parameters
+        ----------
+        function : Callable
+          Function called on every boundary center point to check if it is on
+          the boundary, returns bool-type
+        only_unassigned : bool
+          Uses only previously unassigned boundaries
+          (i.e. on boundary 1)
+        boundary_id : int
+          boundary_id to be assigned. If not chosen set to new lowest value
+
+        Returns
+        -------
+        None
+        """
+        # Get minimum boundary id
+        max_BID = self.interfaces.min()
+
+        if boundary_id is None:
+            new_BID = max_BID - 1
+            log.debug(f"Creating new boundary with ID {abs(new_BID)}")
+        else:
+            # Make sure its negative
+            new_BID = -abs(int(boundary_id))
+            if new_BID < max_BID:
+                log.debug(f"Creating new boundary with ID {abs(new_BID)}")
+            else:
+                log.debug(
+                        "Adding new boundary elements to existing "
+                        f"boundary new boundary with ID {abs(new_BID)}"
+                )
+
+        # retrieve all boundary elements
+        if only_unassigned:
+            boundary_ids = (self.interfaces < 0)
+        else:
+            boundary_ids = self.interfaces == -1
+
+        # Check if there is a boundary
+        if not boundary_ids.any():
+            raise ValueError("No boundary elements could be identified")
+
+        # Check all face centers
+        relevant_boundary_centers = self.spline_boundary_centers[
+                boundary_ids.flatten(), :]
+
+        # Cols and Rows
+        row_ids, col_ids = np.where(boundary_ids)
+
+        # Function is applied to all face centers
+        try:
+            new_boundary_bools = function(relevant_boundary_centers)
+        except BaseException:
+            ValueError(
+                    "Function is not applicable to array. Function layout must"
+                    " f(array<n_face_points,dim>) -> bools<n_face_points>"
+            )
+
+        # Assign new boundary ID to interfaces array
+        self.interfaces[row_ids[new_boundary_bools],
+                        col_ids[new_boundary_bools]] = new_BID
+
+    def add_boundary_from_seed(self, seed_position, tolerance=None):
+        """WIP
+        Starting from a seed position, the splines are propagated until they
+        reach a kink (no g1 continuity). This uses the spline boundary
+        information and determines the interface information.
+
+        Parameters
+        ----------
+        seed_position : array-like (n_dim)
+          Seed position from where the propagation starts
+        tolerance : double
+          Maximum angle (rad) between two normal vectors on boundary which is
+          accepted as g1
+        """
+        from splinepy.settings import TOLERANCE
+
+        if tolerance is None:
+            tolerance = TOLERANCE
+
+        pass
```

Comment:
I remember somebody nagging about having unused part of the coded included

Reply:
<3 not yet used